### PR TITLE
Auth Improvements & tests

### PR DESCRIPTION
**Pull request by lucasvo**
`governance`
uint256 consistency
require message consistency (`TinlakeMgr`)
rwaToken & urn now only set via file

### CODE REVIEW

File: src/mgr.sol [hunk 1]
```diff
@@ -192,54 +188,49 @@ contract TinlakeManager {
 
     // moves the rwaToken into the vault
     // requires that mgr contract holds the rwaToken
-    function lock(uint wad) public operatorOnly {
+    function lock(uint256 wad) public auth {
         require(vat.live() == 1, "TinlakeManager/mkr-in-ES");
+        rwaToken.approve(address(urn), uint256(wad));
```
Comment (lucasvo) [inline src/mgr.sol:196]: Doing this here means we can call `file("rwaToken", addr)` before the `urn` is set. Slightly more robust

File: src/mgr.sol [hunk 2]
```diff
@@ -192,54 +188,49 @@ contract TinlakeManager {
...
         urn.lock(wad);
-        live = true;
```
Comment (lucasvo) [inline src/mgr.sol:198]: any subsequent call will fail without calling lock first but moving it up into the modifier makes more sense as it doesn't create some weird pre-launch state that is now also tracked via `live`

File: src/mgr.sol [hunk 3]
```diff
@@ -248,20 +239,22 @@ contract TinlakeManager {
 
     function file(bytes32 what, address data) public auth {
         emit File(what, data);
-        if (what == "daiJoin") daiJoin = JoinLike(data);
-        else if (what == "end")  end = EndLike(data);
```
Comment (lucasvo) [inline src/mgr.sol:252]: any reason to include this?

File: src/mgr.sol [hunk 4]
```diff
@@ -76,12 +76,10 @@ contract TinlakeManager {
     // --- Auth ---
     mapping (address => uint256) public wards;
     function rely(address usr) external auth {
-        require(live, "TinlakeMgr/not-live");
```
Comment (xmxanuel) [inline src/mgr.sol:79]: good catch

File: src/mgr.sol [hunk 5]
```diff
@@ -140,34 +137,33 @@ contract TinlakeManager {
...
-                address end_,      bytes32 ilk_
+    constructor(address dai_,        address daiJoin_,
+                address drop_,       address pool_,
+                address governance_, address owner_,
+                address tranche_,    address end_,
+                bytes32 ilk_
                 ) public {
 
         dai = GemLike(dai_);
+        daiJoin = JoinLike(daiJoin_);
```
Comment (xmxanuel) [inline src/mgr.sol:152]: you are right, we need it in the recover method

**Review by xmxanuel — APPROVED** [review]
lgtm

File: src/mgr.sol [hunk 6]
```diff
@@ -90,8 +88,11 @@ contract TinlakeManager {
         _;
     }
 
-    modifier operatorOnly {
-        require(msg.sender == operator, "TinlakeMgr/operator-only");
+    // Maker Governance is authorized call tell, unwind, and migrate.
+    address public governance;
+
```
Comment (MrChico) [inline src/mgr.sol:95]: I would make this multiuser as well. Can use the `bud` + `kiss` + `diss` https://github.com/livnev/rwa-example/blob/master/src/RwaConduit.sol#L93 memes

File: src/mgr.sol [hunk 7]
```diff
@@ -72,16 +74,14 @@ interface MIP21UrnLike {
 // not only DROP as an ERC20 balance in this contract, but also what's currently
 // undergoing redemption from the Tinlake pool.
 
-contract TinlakeManager {
+contract TinlakeManager is DSTest {
```
Comment (lucasvo) [inline src/mgr.sol]:
```suggestion
contract TinlakeManager {
```

**Pull request merged**